frontend/expr_inferrer.py

```diff
@@ -124,6 +128,14 @@ def _infer_add(left_type, right_type, lineno, solver):
     result_type = solver.new_z3_const("addition_result")
     solver.add(axioms.add(left_type, right_type, result_type, solver.z3_types),
                fail_message="Addition in line {}".format(lineno))
+
+    # Add soft constraint in case of list addition
```

Comment:
The subtype will be invoked anyway because the soft-constraints are only taken into consideration if the hard ones are satisfied. Moreover, to add a soft constraint for numbers addition, we will need a condition that the operand is a number (which is done by calling `subtype` function). That's why I think this will not give better accuracy and it will to more performance overhead. What do you think?